apps/web/components/NavigationBar/HideOnScroll.tsx

```diff
@@ -0,0 +1,25 @@
+"use client"
+
+import { useEffect, useState } from "react"
+import { cn } from "utils/cn"
+
+export function HideOnScroll({ children }) {
+  const [lastScrollTop, setLastScrollTop] = useState(0)
+  const [hidden, setHidden] = useState(false)
+
+  useEffect(() => {
+    function handleScroll() {
```

Comment:
Would it be possible to do with just position sticky? If not, maybe with intersection observer?

Having scroll listener will be expensive, especially if it isn't throttled. If the two above are not possible in this case then we can try throttling every 16ms (60fps)